frontend/packages/pipelines-plugin/src/components/repository/repository-form-utils.ts

```diff
@@ -0,0 +1,99 @@
+import * as GitUrlParse from 'git-url-parse';
+import { TFunction } from 'i18next';
+import { Base64 } from 'js-base64';
+import * as yup from 'yup';
+import { gitUrlRegex } from '@console/dev-console/src/components/import/validation-schema';
+import { k8sCreateResource } from '@console/dynamic-plugin-sdk/src/utils/k8s';
+import { SecretType } from '@console/internal/components/secrets/create-secret';
+import { SecretModel } from '@console/internal/models';
+import { SecretKind } from '@console/internal/module/k8s';
+import { nameRegex } from '@console/shared/src';
+import { RepositoryModel } from '../../models';
+
+export const repositoryValidationSchema = (t: TFunction) =>
+  yup.object().shape({
+    name: yup
+      .string()
+      .matches(nameRegex, {
+        message: t(
+          'pipelines-plugin~Name must consist of lower-case letters, numbers and hyphens. It must start with a letter and end with a letter or number.',
+        ),
+        excludeEmptyString: true,
+      })
+      .required(t('pipelines-plugin~Required')),
+    gitUrl: yup
+      .string()
+      .matches(gitUrlRegex, t('pipelines-plugin~Invalid Git URL.'))
+      .required(t('pipelines-plugin~Required')),
+    accessToken: yup.string(),
+  });
+
+const createTokenSecret = async (repositoryName: string, token: string, namespace: string) => {
+  const s4 = () =>
+    Math.floor((1 + Math.random()) * 0x10000)
+      .toString(16)
+      .substring(1);
+
+  const data: SecretKind = {
+    apiVersion: SecretModel.apiVersion,
+    kind: SecretModel.kind,
+    metadata: {
+      name: `${repositoryName}-token-${s4()}`,
+    },
```

Comment:
Small improvement idea: Kubernetes supports a special metadata `generateName` attributes to create random suffixed resources. You can simplify this like this:

```suggestion
  const data: SecretKind = {
    apiVersion: SecretModel.apiVersion,
    kind: SecretModel.kind,
    metadata: {
      generateName: `${repositoryName}-token-`,
    },
```

Wdyt?

Reply:
Thanks, did not know about this.